index/scorch/segment/zap/segment.go

```diff
@@ -355,3 +374,20 @@ func (s *Segment) DictAddr(field string) (uint64, error) {
 
 	return s.fieldsOffsets[fieldID-1], nil
 }
+
+func (s *Segment) loadDvIterators() error {
+	if s.docValueOffset == fieldNotUninverted || s.docValueOffset == 0 {
```

Comment:
why is 0 an invalid location of the docValueOffset?

Reply:
assumption was that it can't begin at zero address as there are other valid entries before that at that address location?

Reply:
i think thats true today, but depends on other things unrelated to the implementation of the doc values themselves.  for example, if we decided to move the stored field data to a different section of the file, this might be first and always start at zero, and no one would think to check for this line.  i would prefer that this code not make this assumption.